mars-rover/mars_rover.rb

```diff
@@ -1,3 +1,121 @@
+require_relative 'rover_coordinate'
+
+$NORTH = 'n'
+$EAST  = 'e'
+$SOUTH = 's'
+$WEST  = 'w'
+$STEP_FORWARD  = 'f'
+$STEP_BACKWARD = 'b'
+$TURN_RIGHT    = 'r'
+$TURN_LEFT     = 'l'
+
 class MarsRover
+  attr_accessor :obstacle
+
+  def initialize (current_coordinate,direction,grid)
+    @current_coordinate = current_coordinate
+    @facing = RoverDirection.new(direction)
+    @grid = grid
+  end
+
+  def get_position
+    [@current_coordinate.to_s,@facing.to_s]
+  end
+
+  def execute commands
+    @obstacle = nil
+
+    commands.downcase.each_char do |command|
+      cmd=RoverCommand.new(command)
+      if cmd.step_command? then
+        delta_coordinate = RoverCoordinate.coordinate_from_vector(@facing.to_s)
+
+        next_coordinate = @current_coordinate + delta_coordinate if cmd.step_forward?
+        next_coordinate = @current_coordinate - delta_coordinate if cmd.step_backward?
 
+        next_coordinate = @grid.reposition(next_coordinate) unless @grid.on_grid?next_coordinate
+
```

Comment:
Perhaps the wrapping should be hidden by the grid class? One might pass in a coordinate and vector to the grid class and get a new coordinate back?